internal/armada/server/submit_to_log.go

```diff
@@ -245,6 +245,48 @@ func (srv *PulsarSubmitServer) SubmitJobs(ctx context.Context, req *api.JobSubmi
 }
 
 func (srv *PulsarSubmitServer) CancelJobs(ctx context.Context, req *api.JobCancelRequest) (*api.CancellationResult, error) {
+
+	// For requests specifying a job id, we need to check which queue the job is part of
+	// (see comments below).
+	// This must be done before checking authroization, since the auth check expects a queue.
+	if req.JobId != "" {
+		jobs, err := srv.SubmitServer.jobRepository.GetJobsByIds([]string{req.JobId})
+		if err != nil {
+			return nil, err
+		}
+		if len(jobs) != 1 { // Internal error; should never happen.
```

Comment:
This would happen if the user sends us some crazy job id.  It might be worth distinguishing between jobNotFound (which is probably their fault) and multiple jobs found (which is definitely our fault).

Reply:
I agree. Fixed.